src/trip/components/tripsModal.jsx

```diff
@@ -1,70 +1,103 @@
-import React, { useEffect, useState } from 'react'
-
+import React, { useEffect, useState, useContext } from 'react'
 import { Button, Icon, Modal, Form, Input, Dropdown, Message } from 'semantic-ui-react'
+
+import { TripContext } from '../TripContext'
 import dateFormat from '../../utils/dateFormat'
 
 const TripsModal = ({ mode, initialData, disabled, floated, primary, isBack }) => {
-    
-    const DATA = [
-        { value: "Baptiste", text: "Baptiste" },
-        { value: "Romain", text: "Romain" },
-        { value: "Othmane", text: "Othmane" },
-        { value: "Julie", text: "Julie" },
-        { value: "Hugo", text: "Hugo" },
-        { value: "Patrick", text: "Patrick" },
-        { value: "Léa", text: "Léa" },
-        { value: "Tanguy", text: "Tanguy" },
-        { value: "Momo", text: "Momo" }
-    ]
+    const { login, users, createTrip, updateTrip, getUserByUsername } = useContext(TripContext)
 
+    const [modalOpen, setModalOpen] = useState(undefined)
     const [btnEnabled, setBtnEnabled] = useState(false)
-    const [inputs, setInputs] = useState({ adress: '', seats: 0, date: '', time: '', passengers: [] })
+    const [inputs, setInputs] = useState({ location: '', seats: 1, date: '', time: '', passengers: [] })
     const [error, setError] = useState({ header: null, content: null, isHidden: true })
 
-    useEffect(() => {
+
+    const setOrResetInputs = () => {
         if (initialData) {
-            const date = dateFormat(initialData.start).split(' ')[0]
-            const time = dateFormat(initialData.start).split(' ')[1]
-            const passengers = initialData.passengers.map(passenger => passenger.username)
-            return setInputs({ adress: initialData.adress, seats: initialData.seats, date, time, passengers })
+            const date = dateFormat(initialData.date).split(' ')[0]
+            const time = dateFormat(initialData.date).split(' ')[1]
+            const { _id, driver, location, passengers, seats, type } = initialData
+            return setInputs({ _id, date, time, driver, location, passengers, seats, type })
+        } else {
+            return setInputs({ location: '', seats: 1, date: '', time: '', passengers: [] })
         }
-    },[])
+    }
+
+    useEffect(() => setOrResetInputs(), [])
 
-    useEffect(() => {        
+    useEffect(() => {
         let emptyFields = false
         
-        Object.values(inputs).forEach(value => { if (typeof value !== "object" && !value) emptyFields = true; })
+        for (const key in inputs) {
```

Comment:
You could use `Object.entries` to get keys and values :)

Reply:
Done